src/api_types.jl

```diff
@@ -0,0 +1,310 @@
+## C types
+const C_time_t = Int
+
+## HDF5 types and constants
+const haddr_t  = UInt64
+const herr_t   = Cint
+const hid_t    = Int64
+const hsize_t  = UInt64
+const hssize_t = Int64
+const htri_t   = Cint   # pseudo-boolean (negative if error)
+
+# For VLEN
+struct hvl_t
+    len::Csize_t
+    p::Ptr{Cvoid}
+end
+const HVL_SIZE = sizeof(hvl_t)
+
+# Object reference types
+struct HDF5ReferenceObj
+    r::UInt64 # Size must be H5R_OBJ_REF_BUF_SIZE
+end
+const HDF5ReferenceObj_NULL = HDF5ReferenceObj(UInt64(0))
+
+# For group information
+struct H5G_info_t
+    storage_type::Cint # enum H5G_storage_type_t
+    nlinks::hsize_t
+    max_corder::Int64
+    mounted::Cuint
+end
+
+# For objects
+struct H5_ih_info_t
+    index_size::hsize_t
+    heap_size::hsize_t
+end
+struct H5O_info_t # version 1 type H5O_info1_t
+    fileno::Cuint
+    addr::haddr_t
+    otype::Cint # enum H5O_type_t
+    rc::Cuint
+    atime::C_time_t
```

Comment:
`C_time_t` -> `Ctime_t` to match `Csize_t` et al.  in base julia?

Reply:
Yeah, I think that'd be a good change.